operator/internal/handlers/internal/tlsprofile/tlsprofile_test.go

```diff
@@ -0,0 +1,120 @@
+		"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+		"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+		"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+		"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256",
+		"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256",
+		"TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
+		"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
+		"TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
+		"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
+		"TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA",
+		"TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
+		"TLS_RSA_WITH_AES_128_GCM_SHA256",
+		"TLS_RSA_WITH_AES_256_GCM_SHA384",
+		"TLS_RSA_WITH_AES_128_CBC_SHA256",
+		"TLS_RSA_WITH_AES_128_CBC_SHA",
+		"TLS_RSA_WITH_AES_256_CBC_SHA",
+		"TLS_RSA_WITH_3DES_EDE_CBC_SHA",
+	}
+	ciphersIntermediate = []string{
+		"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+		"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+		"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+		"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+		"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256",
+		"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256",
+	}
+)
+
+func TestGetSecurityProfileInfo_Old(t *testing.T) {
+	k := setupClient()
+
+	// Old Profile
+	profile := projectconfigv1.TLSProfileOldType
+	expected := projectconfigv1.TLSProfileSpec{
+		MinTLSVersion: "VersionTLS10",
+		Ciphers:       ciphersOld,
+	}
+
+	info, err := tlsprofile.GetSecurityProfileInfo(context.TODO(), k, profile)
+	assert.Nil(t, err)
+	assert.NotNil(t, info)
+	assert.EqualValues(t, expected, info)
+}
+
+func TestGetSecurityProfileInfo_Intermediate(t *testing.T) {
+	k := setupClient()
+
+	// Intermediate Profile
+	profile := projectconfigv1.TLSProfileIntermediateType
+	expected := projectconfigv1.TLSProfileSpec{
+		MinTLSVersion: "VersionTLS12",
+		Ciphers:       ciphersIntermediate,
+	}
+
+	info, err := tlsprofile.GetSecurityProfileInfo(context.TODO(), k, profile)
+	assert.Nil(t, err)
+	assert.NotNil(t, info)
+	assert.EqualValues(t, expected, info)
+}
+
+func TestGetSecurityProfileInfo_Modern(t *testing.T) {
+	k := setupClient()
+
+	// Modern Profile
+	profile := projectconfigv1.TLSProfileModernType
+	expected := projectconfigv1.TLSProfileSpec{
+		MinTLSVersion: "VersionTLS13",
+		// Go lib crypto doesn't allow ciphers to be configured for TLS 1.3
+		// (Read this and weep: https://github.com/golang/go/issues/29349)
+		Ciphers: []string{},
+	}
+
+	info, err := tlsprofile.GetSecurityProfileInfo(context.TODO(), k, profile)
+	assert.Nil(t, err)
+	assert.NotNil(t, info)
+	assert.EqualValues(t, expected, info)
+}
+
+func setupClient() *k8sfakes.FakeClient {
+	sw := &k8sfakes.FakeStatusWriter{}
+	k := &k8sfakes.FakeClient{}
+
+	k.GetStub = func(_ context.Context, name types.NamespacedName, object client.Object) error {
+		if apiServer.Name == name.Name {
+			k.SetClientObject(object, &apiServer)
+			return nil
+		}
+		return apierrors.NewNotFound(schema.GroupResource{}, "something wasn't found")
+	}
+
+	k.StatusStub = func() client.StatusWriter { return sw }
+
+	return k
+}
```

Comment:
I suggest not to create such helpers as the test cases should IMHO code wide isolated from each other. If you want to write the client once the pattern is to write a test table instead of three test functions.